### DISCUSSION TITLE
How to upgrade cloudflare to react 18 w/streaming?

### DISCUSSION THREAD

**Discussion by DevOfManyThings** [remix-run/remix#4441] [Q&A]
Currently the `cloudflare-workers` and `cloudflare-pages` packages use react 17, and `renderToString` instead of react 18 with `renderToPipeableStream`.

Cloudflare released the ability to set compatibility flags on pages recently, so it's now possible to upgrade through setting the `streams_enable_constructors` & `transformstream_enable_standard_constructor` flags, this is how they [announced support for nextjs](https://blog.cloudflare.com/next-on-pages/), setting these flags will not be necessary after 30th of November when they become default.

I think this needs updates to some of the documentation and templates, but I'm not sure if anyone else has started work on this?

**Comment by lili21** [comment]
I've created a PR for this. https://github.com/remix-run/remix/pull/4502
